sky/backends/wheel_utils.py

```diff
@@ -161,12 +161,19 @@ def _get_latest_modification_time(path: pathlib.Path) -> float:
         last_modification_time = _get_latest_modification_time(SKY_PACKAGE_PATH)
         last_wheel_modification_time = _get_latest_modification_time(WHEEL_DIR)
 
-        # only build wheels if the wheel is outdated
-        if last_wheel_modification_time < last_modification_time:
+        # Only build wheels if the wheel is outdated or wheel does not exist
+        # for the requested version.
+        if (last_wheel_modification_time < last_modification_time) or not any(
+                WHEEL_DIR.glob(_WHEEL_PATTERN)):
```

Comment:
Should we instead check `WHEEL_DIR.glob(f'**/{_WHEEL_PATTERN}')` instead, similar to https://github.com/skypilot-org/skypilot/blob/master/sky/backends/wheel_utils.py#L43?

Also, since we are checking the whether the current wheel exists, it might be fine to just use the current wheel for the later `latest_wheel`, i.e. it can become `_get_current_wheel_and_remove_older_others`?

Reply:
Ah good idea. Refactored `_get_latest_wheel_and_remove_all_others` into separate methods to get the latest wheel and remove others.